genesyscloud/architect_user_prompt/genesyscloud_architect_user_prompt_proxy.go

```diff
@@ -0,0 +1,171 @@
+	return p.updateArchitectUserPromptAttr(ctx, p, id, body)
+}
+
+// deleteArchitectUserPrompt deletes a user prompt
+func (p *architectUserPromptProxy) deleteArchitectUserPrompt(ctx context.Context, id string, allResources bool) (*platformclientv2.APIResponse, error) {
+	return p.deleteArchitectUserPromptAttr(ctx, p, id, allResources)
+}
+
+// createArchitectUserPromptResource creates a new user prompt resource
+func (p *architectUserPromptProxy) createArchitectUserPromptResource(ctx context.Context, id string, body platformclientv2.Promptassetcreate) (*platformclientv2.Promptasset, *platformclientv2.APIResponse, error) {
+	return p.createArchitectUserPromptResourceAttr(ctx, p, id, body)
+}
+
+// updateArchitectUserPromptResource updates a user prompt resource
+func (p *architectUserPromptProxy) updateArchitectUserPromptResource(ctx context.Context, id string, languageCode string, body platformclientv2.Promptasset) (*platformclientv2.Promptasset, *platformclientv2.APIResponse, error) {
+	return p.updateArchitectUserPromptResourceAttr(ctx, p, id, languageCode, body)
+}
+func createArchitectUserPromptFn(ctx context.Context, p *architectUserPromptProxy, body platformclientv2.Prompt) (*platformclientv2.Prompt, *platformclientv2.APIResponse, error) {
+	prompt, response, err := p.architectApi.PostArchitectPrompts(body)
+	if err != nil {
+		return nil, response, fmt.Errorf("failed to create architect user prompt: %s", err)
+	}
+	return prompt, response, nil
+}
+
+func getArchitectUserPromptFn(ctx context.Context, p *architectUserPromptProxy, id string, includeMediaUris bool, includeResources bool, language []string) (*platformclientv2.Prompt, *platformclientv2.APIResponse, error, bool) {
+	prompt, response, err := p.architectApi.GetArchitectPrompt(id, includeMediaUris, includeResources, language)
+	if err != nil {
+		return nil, response, fmt.Errorf("failed to retrieve architect user prompt: %s", err), true
+	}
+	return prompt, response, nil, false
+}
+
+func updateArchitectUserPromptFn(ctx context.Context, p *architectUserPromptProxy, id string, body platformclientv2.Prompt) (*platformclientv2.Prompt, *platformclientv2.APIResponse, error) {
+	prompt, response, err := p.architectApi.PutArchitectPrompt(id, body)
+	if err != nil {
+		return nil, response, fmt.Errorf("failed to update architect user prompt: %s", err)
+	}
+	return prompt, response, nil
+}
+
+func deleteArchitectUserPromptFn(ctx context.Context, p *architectUserPromptProxy, id string, allResources bool) (*platformclientv2.APIResponse, error) {
+	response, err := p.architectApi.DeleteArchitectPrompt(id, allResources)
+	if err != nil {
+		return response, fmt.Errorf("failed to delete architect user prompt: %s", err)
+	}
+	return response, nil
+}
+
+func getArchitectUserPromptsFn(ctx context.Context, p *architectUserPromptProxy, includeMediaUris bool, includeResources bool, nameArr []string) (*[]platformclientv2.Prompt, *platformclientv2.APIResponse, error, bool) {
```

Comment:
Are the values of includeMediaUris, includeResources, nameArr every anything other than false and nil? If not you can remove them from here and just set them directly on the function call below

Reply:
they do  differ